README.md

```diff
@@ -36,6 +36,11 @@ The class provides a default implementation of `_transform` which will use a gen
 `*_inflate` to generate chunks of data to be pushed from a chunk that is written to the stream.
 Subclasses must override `*_inflate`, or provide it via the constructor option `inflate`.
 
+To accommodate generators that need to perform asynchronous work to transform a chunk,
+generator methods in this class can yield Promises. The class will wait for the Promise to
+resolve before pushing the value. If the Promise rejects, the error will be passed to the
+transform callback function.
```

Comment:
This isn't what you've actually implemented, though if it were I, I'd support both.

However, there is a difference between yielding a promise from a generator (which will return `next.done` as `false` and `next.value` as a Thenable, which will resolve to a value), and an async generator (which will return `next` itself as a Thenable, which will resolve to an object with `object.done` and `object.value` set). This comment describes the former, but you've implemented the latter.

As I say, I don't think it's a big code change to support both, so I'd do that, and document both as well. 😄 But probably get one right first. 😉

Reply:
Fair point. It's to do with how the test was implemented, where as we need two test cases.

Reply:
I haven't looked at the tests properly yet. Probably wait for another revision. It took a while to build a mental model adequate to understand what the actual code was doing (mostly because of the issues I raised in the overall comment).